### PR TITLE
Add tooltips for `SegmentedButton`

### LINKED ISSUE #128374: [Feature Request] Add Tooltips for SegmentedButton
# Issue
When using `SegmentedButton` you might want to add tooltips explaing each provided option. Right now you can either add `Tooltip` around whole `SegmentedButton` which won't allow you to have separate tooltips for each segment or you can add `Tooltip` as a `label` of each `ButtonSegment`, but then only the label of each segment will have tooltip and not the whole segment (which also includes icon and padding around them)
# Proposal
Add `String? tooltip` field for `ButtonSegment` and if present, put `Tooltip` around corresponding segment widget

### PR DESCRIPTION
**Pull request by ZhulanovAA**
Add `String? tooltip` field for `ButtonSegment`. When present, put `Tooltip` around corresponding segment of `SegmentedButton`.

Fixes #128374

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/segmented_button.dart [hunk 1]
```diff
@@ -335,11 +340,18 @@ class SegmentedButton<T> extends StatelessWidget {
             child: label,
           );
 
+      final Widget buttonWithTooltip = (segment.tooltip != null)
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/segmented_button.dart:343]: are the parentheses needed?

Comment (ZhulanovAA) [reply packages/flutter/lib/src/material/segmented_button.dart:343]: No, I've completely missed the fact they are not obligatory in Dart. I don't have much experience with contributing, how shuld this be fixed now, when it's merged already?

**Comment by HansMuller** [comment]
It you want to submit another PR without the parens, we'll land it. Doesn't need tests

**Comment by ZhulanovAA** [comment]
Should a new issue be opened for a new PR or should it mention the same closed issue?

While adding tests I've noticed a couple of old `SegmentedButton` tests are missing indent and there are also one unnecessary empty line there. Will it be ok to add a new issue "Clean up `SegmentedButton` code" and fix both my and old code together?

**Comment by chunhtai** [comment]
no issue is fine, just open a new pr and pin Hans or me on it

**Pull request merged**